### PR TITLE
feat: cookie theme toggler

### PR DESCRIPTION
**Pull request by solidovic**

### CODE REVIEW

File: packages/cookie-theme-toggler/src/cookie-theme-provider.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      topLevelDomain = 'localhost'
+    } else {
+      // Get host with dot in first position
+      topLevelDomain = `.${document.location.host
+        .split('.')
+        .slice(-2)
+        .join('.')}`
+
+      // Remove port
+      topLevelDomain = topLevelDomain.split(':').slice(0, 1).join()
```
Comment (Jabher) [inline packages/cookie-theme-toggler/src/cookie-theme-provider.tsx:53]: just use hostname instead

File: packages/cookie-theme-toggler/src/styles.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import styled from 'styled-components'
+import { Button } from '@lidofinance/button'
+
+export const CookieThemeTogglerStyle = styled((props) => (
+  <Button {...props} />
+)).attrs({
```
Comment (Jabher) [inline packages/cookie-theme-toggler/src/styles.tsx:6]: I'm unsure but I remember you could just write `styled(Button)` for this case

File: packages/cookie-theme-toggler/src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  const match = document.cookie.match(
+    new RegExp('(^| )' + COOKIE_THEME_MANUAL_KEY + '=([^;]+)')
+  )
+
+  if (!match) {
+    return null
+  }
+
+  return match[2] as ThemeName
+}
```
Comment (Jabher) [inline packages/cookie-theme-toggler/src/utils.ts:14]: 1. should not be some `typeof document` validation here?
2. please double-check, as I can be missing something 

```suggestion
const cookieThemeMatcher = new RegExp(`(^| )${COOKIE_THEME_MANUAL_KEY}=([^;]+)`)
export const getThemeNameFromCookies = (): ThemeName | null =>
  document.cookie.match(cookieThemeMatcher)?.[2] ?? null
```

Comment (solidovic) [reply packages/cookie-theme-toggler/src/utils.ts:14]: We use this function in 'if (typeof window !== 'undefined') {'
https://github.com/lidofinance/ui/pull/346/files#diff-377f43adac8a0c6a51951621d47a102b023109a07ea5400b35620f9b60e0ecd0R40

but I agree with you that we should add typeof document validation here. I will make that

File: packages/cookie-theme-toggler/src/styles.tsx (hunk 2 above)
Comment (solidovic) [reply packages/cookie-theme-toggler/src/styles.tsx:6]: Yes. But I dont know someone will be use props like size btn and other (my suggest is no - so I will remove props)

**Review by Jabher — APPROVED** [review]

**Pull request merged**